LoopKitUI/Views/Settings Editors/TherapySettingsViewModel.swift

```diff
@@ -111,3 +116,58 @@ public class TherapySettingsViewModel: ObservableObject {
         didSave?(TherapySetting.insulinSensitivity, therapySettings)
     }
 }
+
+// MARK: Navigation
+
+extension TherapySettingsViewModel {
+
+    func screen(for setting: TherapySetting) -> (_ goBack: @escaping () -> Void) -> AnyView {
+        switch setting {
+        case .suspendThreshold:
+            return { goBack in
+                AnyView(SuspendThresholdEditor(therapySettingsViewModel: self, didSave: goBack).environment(\.dismiss, goBack))
+            }
+        case .glucoseTargetRange:
+            return { goBack in
+                AnyView(CorrectionRangeScheduleEditor(viewModel: self, didSave: goBack).environment(\.dismiss, goBack))
+            }
+        case .preMealCorrectionRangeOverride:
+            return { goBack in
+                AnyView(CorrectionRangeOverridesEditor(viewModel: self, preset: .preMeal, didSave: goBack).environment(\.dismiss, goBack))
+            }
+        case .workoutCorrectionRangeOverride:
+            return { goBack in
+                AnyView(CorrectionRangeOverridesEditor(viewModel: self, preset: .workout, didSave: goBack).environment(\.dismiss, goBack))
+            }
+        case .basalRate:
+            if self.pumpSupportedIncrements?() != nil {
+                return { goBack in
+                    AnyView(BasalRateScheduleEditor(viewModel: self, didSave: goBack).environment(\.dismiss, goBack))
+                }
+            }
+        case .deliveryLimits:
+            if self.pumpSupportedIncrements?() != nil {
+                return { goBack in
+                    AnyView(DeliveryLimitsEditor(viewModel: self, didSave: goBack).environment(\.dismiss, goBack))
+                }
+            }
+        case .insulinModel:
+            if self.therapySettings.insulinModelSettings != nil {
+                return { goBack in
+                    AnyView(InsulinModelSelection(viewModel: self, didSave: goBack).environment(\.dismiss, goBack))
+                }
+            }
+        case .carbRatio:
+            return { goBack in
+                AnyView(CarbRatioScheduleEditor(viewModel: self, didSave: goBack).environment(\.dismiss, goBack))
+            }
+        case .insulinSensitivity:
+            return { goBack in
+                return AnyView(InsulinSensitivityScheduleEditor(viewModel: self, didSave: goBack).environment(\.dismiss, goBack))
+            }
+        case .none:
+            break
+        }
+        return { _ in AnyView(Text("\(setting.title)")) }
```

Comment:
[nit] Not sure I understand the default return here.

Reply:
Should never happen...it would be a coding error. Could be `fatalError` but that seemed draconian, so it just shows the setting name in the middle of the screen that is not covered.